pkg/cache/clusterqueue_test.go

```diff
@@ -367,3 +368,165 @@ func TestClusterQueueUpdate(t *testing.T) {
+			admissionChecks: map[string]AdmissionCheck{
+				"check1": {
+					Active:     true,
+					Controller: "controller1",
+				},
+				"check2": {
+					Active:     true,
+					Controller: "controller2",
+				},
+			},
+			singleInstanceControllers: []string{"controller1", "controller2"},
+			wantStatus:                pending,
+			wantReason:                "CheckNotFoundOrInactive",
+		},
+		{
+			name:     "Active clusterQueue updated with inactive AC",
+			cqStatus: active,
+			admissionChecks: map[string]AdmissionCheck{
+				"check1": {
+					Active:     true,
+					Controller: "controller1",
+				},
+				"check2": {
+					Active:     true,
+					Controller: "controller2",
+				},
+				"check3": {
+					Active:     false,
+					Controller: "controller3",
+				},
+			},
+			singleInstanceControllers: []string{"controller1", "controller2"},
+			wantStatus:                pending,
+			wantReason:                "CheckNotFoundOrInactive",
+		},
+		{
+			name:     "Active clusterQueue updated with duplicate single instance AC Controller",
+			cqStatus: active,
+			admissionChecks: map[string]AdmissionCheck{
+				"check1": {
+					Active:     true,
+					Controller: "controller1",
+				},
+				"check2": {
+					Active:     true,
+					Controller: "controller2",
+				},
+				"check3": {
+					Active:     true,
+					Controller: "controller2",
+				},
+			},
+			singleInstanceControllers: []string{"controller1", "controller2"},
+			wantStatus:                pending,
+			wantReason:                "MultipleSingleInstanceControllerChecks",
+		},
+		{
+			name:     "Terminating clusterQueue updated with valid AC list",
+			cqStatus: terminating,
+			admissionChecks: map[string]AdmissionCheck{
+				"check1": {
+					Active:     true,
+					Controller: "controller1",
+				},
+				"check2": {
+					Active:     true,
+					Controller: "controller2",
+				},
+				"check3": {
+					Active:     true,
+					Controller: "controller3",
+				},
+			},
+			wantStatus: terminating,
+			wantReason: "Terminating",
+		},
+		{
+			name:     "Terminating clusterQueue updated with not found AC",
+			cqStatus: terminating,
+			admissionChecks: map[string]AdmissionCheck{
+				"check1": {
+					Active:     true,
+					Controller: "controller1",
+				},
+				"check2": {
+					Active:     true,
+					Controller: "controller2",
+				},
+			},
+			wantStatus: terminating,
+			wantReason: "Terminating",
+		},
+	}
+
+	for _, tc := range testcases {
+		t.Run(tc.name, func(t *testing.T) {
+			cache := New(utiltesting.NewFakeClient())
+			cq, err := cache.newClusterQueue(cq)
+			if err != nil {
+				t.Fatalf("failed to new clusterQueue %v", err)
+			}
+
+			cq.Status = tc.cqStatus
+
+			// Align the admission check related internals to the desired Status.
+			if tc.cqStatus == active {
+				cq.hasMultipleSingleInstanceControllersChecks = false
+				cq.hasMissingOrInactiveAdmissionChecks = false
+			} else {
+				cq.hasMultipleSingleInstanceControllersChecks = true
```

Comment:
I'm still puzzled by this, why status != active implies `hasMultipleSingleInstanceControllersChecks: true`, could this be inactive for other reasons, right? This way we cannot test what happens if inactive for reason A, but both `has` are true (or one of them).

Reply:
both of them are reevaluated during the call to cq.updateWithAdmissionChecks.

Reply:
I see, sgtm